apps/src/jsArray.js

```diff
@@ -0,0 +1,113 @@
+// Strict linting: Absorb into global config when possible
+/* jshint
+ unused: true,
+ eqeqeq: true,
+ maxlen: 120
+ */
+
+var codegen = require('./codegen');
+
+// Array.sort
+// iterative quicksort based on:
+// http://www.stoimen.com/blog/2010/06/18/friday-algorithms-iterative-quicksort/
+// (modified as a stateful native function to support callbacks)
+exports.generateSort = function (jsInterpreter) {
+  return function (opt_compFunc) {
```

Comment:
Might be worth giving the returned function a comment describing what it does (sorts an array using quicksort, in place?), what the optional param is (a function that takes two args and returns ___).